charger/ocpp.go

```diff
@@ -347,38 +365,54 @@ func (c *OCPP) setChargingProfile(connectorId int, profile *types.ChargingProfil
 	return c.wait(err, rc)
 }
 
-// updatePeriod sets a single charging schedule period with given current and phases
-func (c *OCPP) updatePeriod(current float64, phases int) error {
+// updatePeriod sets a single charging schedule period with given current
+func (c *OCPP) updatePeriod(current float64) error {
 	// current period can only be updated if transaction is active
 	if enabled, err := c.Enabled(); err != nil || !enabled {
 		return err
 	}
 
 	current = math.Trunc(10*current) / 10
 
-	err := c.setChargingProfile(c.connector, getTxChargingProfile(current, phases))
+	err := c.setChargingProfile(c.connector, c.getTxChargingProfile(current))
 	if err != nil {
 		err = fmt.Errorf("set charging profile: %w", err)
 	}
 
 	return err
 }
 
-func getTxChargingProfile(current float64, phases int) *types.ChargingProfile {
-	period := types.NewChargingSchedulePeriod(0, current)
+func (c *OCPP) getTxChargingProfile(current float64) *types.ChargingProfile {
+	var phases = c.phases
+	if phases == 0 {
+		phases = 3
+	}
+
+	var value float64
+	if c.chargingRateUnit == types.ChargingRateUnitWatts {
+		voltage := 230.0
+		// // Get number of currently active phases from Loadpoint
+		// lpPhases := c.lp.GetPhases()
 
-	// TODO add phases support
-	// if phases != 0 {
-	// 	period.NumberPhases = &phases
-	// }
+		// if phases != lpPhases {
+
+		// }
+
+		value = math.Trunc(current*float64(phases)*voltage*10.0) / 10
```

Comment:
Warum?

Reply:
Warum math.Trunc()? Weil beim Strom auch so war. Ich dachte, das wäre nötig.

Reply:
Ich glaube nicht, dass wir Watt auf eine Nachkommastelle brauchen

Reply:
Ach in die Richtung. Ja, ok.